### DISCUSSION TITLE
How to write Unit Tests with TanStack Router?

### DISCUSSION THREAD

**Discussion by Kmsbismarckx** [TanStack/router#1749] [General]
Hello guys,
I did not found a docs for writting unit test with TanStack Router. Is it possible, and how to do that?

**Comment by SeanCassiere** [comment]
We have a unit test setup using vitest and testing-library for the `@tanstack/react-router` package.

https://github.com/TanStack/router/blob/main/packages/react-router/tests/routeContext.test.tsx

You can look at that and get an idea of how you'd want to setup something in your project.
